src/drivers/heater/heater.h

```diff
@@ -0,0 +1,271 @@
+	 * @param argv Pointer to the input argument array.
+	 * @return Returns the heater proportional gain value iff successful, 0.0f otherwise.
+	 */
+	float proportional(char *argv[]);
+
+	/**
+	 * @brief Reports the heater target sensor.
+	 * @return Returns the id of the target sensor
+	 */
+	uint32_t sensor_id();
+
+	/**
+	 * @brief Initiates the heater driver work queue, starts a new background task,
+	 *        and fails if it is already running.
+	 * @return Returns 1 iff start was successful.
+	 */
+	int start();
+
+	/**
+	 * @brief Reports curent status and diagnostic information about the heater driver.
+	 * @return Returns 0 iff successful, -1 otherwise.
+	 */
+	int print_status();
+
+	/**
+	 * @brief Reports the current heater temperature.
+	 * @return Returns the current heater temperature value iff successful, -1.0f otherwise.
+	 */
+	float sensor_temperature();
+
+	/**
+	 * @brief Sets and/or reports the heater target temperature.
+	 * @return Returns the heater target temperature value iff successful, -1.0f otherwise.
+	 */
+	float temperature_setpoint(char *argv[]);
+
+protected:
+
+	/**
+	 * @brief Called once to initialize uORB topics.
+	 */
+	void initialize_topics();
+
+	/**
+	 * @see ModuleBase::initialize_trampoline().
+	 * @brief Trampoline initialization.
+	 * @param arg Pointer to the task startup arguments.
+	 */
+	static void initialize_trampoline(void *arg);
+
+private:
+
+	/**
+	 * @brief Checks for new commands and processes them.
+	 */
+	void process_commands();
+
+	/**
+	 * @brief Trampoline for the work queue.
+	 */
+	static void cycle_trampoline(void *arg);
+
+	/**
+	 * @brief Calculates the heater element on/off time, carries out
+	 *        closed loop feedback and feedforward temperature control,
+	 *        and schedules the next cycle.
+	 */
+	void cycle();
+
+	/**
+	 * @brief Updates the uORB topics for local subscribers.
+	 * @param meta The uORB metadata to copy.
+	 * @param handle The uORB handle to obtain data from.
+	 * @param buffer The data buffer to copy data into.
+	 * @return Returns true iff update was successful.
+	 */
+	int orb_update(const struct orb_metadata *meta, int handle, void *buffer);
+
+	/**
+	 * @brief Updates and checks for updated uORB parameters.
+	 * @param force Boolean to determine if an update check should be forced.
+	 */
+	void update_params(const bool force = true);
+
+	/** @param _command_ack_pub The command ackowledgement topic. */
+	orb_advert_t _command_ack_pub = nullptr;
+
+	/** @param _controller_period_usec The heater controller time period in microseconds.*/
+	int _controller_period_usec;
+
+	/** @param _controller_time_on_usec The heater time on in microseconds.*/
+	int _controller_time_on_usec;
+
+	/** @param _duty_cycle The heater on duty cycle value. */
+	float _duty_cycle;
+
+	/** @param _heater_on Indicator for the heater on/off status. */
+	bool _heater_on;
+
+	/** @param _integrator_value The heater controller integrator value. */
```

Comment:
Incorrect use of `@param`. ` @param _integrator_value` can simply be dropped. Same for all others.

Reply:
How about we use @var instead?

Reply:
I have not seen `@var` being used typically for that, and I'm pretty sure `/** The heater controller integrator value. */` leads to the desired result. Why write more than necessary? Plus there's a redundancy in the variable name being used twice.

Reply:
Hi @bkeung, Here's what's neat about completing all of the doxy comments in a file,... when the docs are generated full visibility into the structure and content of the file is avialable.  For example:
![image](https://user-images.githubusercontent.com/2497951/45056131-46b56c80-b04f-11e8-8f0d-1bca89d00b7e.png)

![image](https://user-images.githubusercontent.com/2497951/45056180-62207780-b04f-11e8-8737-06773dfd087a.png)

When we selectively remove some but not all of the variables it simply paints an incomplete picture in the generated docs.

Reply:
Even the `default Constructor` comment gives a hook for links to be generated in the docs, which also gives visibility to the virtual destrcutor: 
![image](https://user-images.githubusercontent.com/2497951/45057062-dcea9200-b051-11e8-8924-5bd2e19260b1.png)

Reply:
@mcsauder please don't bring up the same point again and again. The discussion is endless, without further arguments being added. The default constructor with the link will also be there without a comment.
Also my comment here is a completely different one: it's about the format, not about removing comments.

Reply:
I understand @bkueng, really I'm not trying to beat a dead horse.  If you look at both screen captures above you'll see that without the tags the constructors are omitted from the docs, that's the only reason for having the tag, and I won't bring it up anymore, I promise.

I am sorry, I still don't think I understand your comment about format.  Without the tag and variable name the docs won't generate any information about the variable, or the constructor, or the method.  Without the tag the documentation is kind of pointless, the variable name can stand on its own if generated docs aren't a concern.  Is the format you prefer to omit the doxy tags completely?  I am sorry, I'm really having a hard time reading what your intent for the documentation is.  Thanks for your patience.

Again, apologies for the inconvenience, I'm not trying to be a pain.